File: Api.Server/src/Center.Service/MessageSendRecord/Dto/QueryMessageSendRecordPagedOutput.cs
// ------------------------------------------------------------------------
// Apache开源许可证
// 
// 版权所有 © 2018-Now 小方
// 
// 许可授权：
// 本协议授予任何获得本软件及其相关文档（以下简称“软件”）副本的个人或组织。
// 在遵守本协议条款的前提下，享有使用、复制、修改、合并、发布、分发、再许可、销售软件副本的权利：
// 1.所有软件副本或主要部分必须保留本版权声明及本许可协议。
// 2.软件的使用、复制、修改或分发不得违反适用法律或侵犯他人合法权益。
// 3.修改或衍生作品须明确标注原作者及原软件出处。
// 
// 特别声明：
// - 本软件按“原样”提供，不提供任何形式的明示或暗示的保证，包括但不限于对适销性、适用性和非侵权的保证。
// - 在任何情况下，作者或版权持有人均不对因使用或无法使用本软件导致的任何直接或间接损失的责任。
// - 包括但不限于数据丢失、业务中断等情况。
// 
// 免责条款：
// 禁止利用本软件从事危害国家安全、扰乱社会秩序或侵犯他人合法权益等违法活动。
// 对于基于本软件二次开发所引发的任何法律纠纷及责任，作者不承担任何责任。
// ------------------------------------------------------------------------

namespace Fast.Center.Service.MessageSendRecord.Dto;

/// <summary>
/// 获取消息发送记录分页列表输出
/// </summary>
public class QueryMessageSendRecordPagedOutput
{
    /// <summary>
    /// 记录Id
    /// </summary>
    public long RecordId { get; set; }

    /// <summary>
    /// 渠道
    /// </summary>
    public MessageSendChannelEnum Channel { get; set; }

    /// <summary>
    /// 收件人
    /// </summary>
    [SugarSearchValue]
    public string Receiver { get; set; }

    /// <summary>
    /// 标题
    /// </summary>
    [SugarSearchValue]
    public string Title { get; set; }

    /// <summary>
    /// 是否成功
    /// </summary>
    public bool IsSuccess { get; set; }

    /// <summary>
    /// 设备
    /// </summary>
    public virtual string Device { get; set; }

    /// <summary>
    /// 操作系统（版本）
    /// </summary>
    public virtual string OS { get; set; }

    /// <summary>
    /// 浏览器（版本）
    /// </summary>
    public virtual string Browser { get; set; }

    /// <summary>
    /// 省份
    /// </summary>
    public virtual string Province { get; set; }

    /// <summary>
    /// 城市
    /// </summary>
    public virtual string City { get; set; }

    /// <summary>
    /// Ip
    /// </summary>
    public virtual string Ip { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    [SugarSearchTime]
    public DateTime? CreatedTime { get; set; }
}

File: App.Client/src/api/services/Center/messageSendRecord/models/QueryMessageSendRecordPagedOutput.ts
import type { MessageSendChannelEnum } from "@/api/enums/MessageSendChannelEnum";

/**
 * 获取消息发送记录分页列表输出
 */
export interface QueryMessageSendRecordPagedOutput {
	/**
	 * 记录Id
	 */
	recordId?: string;
	/**
	 * 
	 */
	channel?: MessageSendChannelEnum;
	/**
	 * 收件人
	 */
	receiver?: string;
	/**
	 * 标题
	 */
	title?: string;
	/**
	 * 是否成功
	 */
	isSuccess?: boolean;
	/**
	 * 设备
	 */
	device?: string;
	/**
	 * 操作系统（版本）
	 */
	os?: string;
	/**
	 * 浏览器（版本）
	 */
	browser?: string;
	/**
	 * 省份
	 */
	province?: string;
	/**
	 * 城市
	 */
	city?: string;
	/**
	 * Ip
	 */
	ip?: string;
	/**
	 * 创建时间
	 */
	createdTime?: string;
}

